### ISSUE TITLE
BinaryValueFunctions are not specific enough

### ISSUE THREAD

**Issue by tm-ms** [microstream-one/microstream#38]
Storing a primitive value from an instance in the heap to the storing buffer and setting a primitive value from the loading buffer to an instance in the heap.

So far, every primitive value size (1, 2, 4, 8) has only 4 function for each category (store, storeReversed, set, setReversed and skip). Because for low level memory operations (Unsafe), this is sufficient and it keeps helping the file small (4+1 + 3+1 + 4+1 + 3+1 + 4 = 22 functions).

However, when using MemoryAccessGeneric, meaning Unsafe operations are replaced by DirectByteBuffer API and Reflection, this is no longer sufficient. Reflection setting- and getting-methods perform a rigid type check: a 1 byte boolean value cannot be read as a 1 byte byte value and so on.

This came up during a more intensive Android test.
(It's a little strange why this didn't come up on the very first test of MemoryAccessorGeneric because the storers are used already during initialization. But whatever ...)

The solution must be to create type-specific storers and setters instead of just byte-size-specific ones.

Meaning 9 storers, 7 reversed storers, 9 setters, 7 reverses setters. The skipping setters can remain at 4 because they just skip n bytes in the binary form data.
Making a total of 36 instead of 22. Not dramatically more, but then the primitive type handling is strictly correct, even for reflection.

**Comment by tm-ms** [comment]
I implemented the additional 14 functions, but I noticed a potential conversion error for reversing functions along the way:

There are no reverseBytes() methods for Float and Double.
This means in order to reverse their bytes, they have to be converted back and forth between a int/long representation.
The problem is that the JavaDoc of Double#longBitsToDouble (and for the Float variant respectively) states: 
>Note that this method may not be able to return a double NaN with exactly same bit pattern as the long argument.

This all of a sudden means:
Where the less specific STORE_8_REVERSED function was absolutely correct for longs and doubles, the more specific STORE_double_REVERSED function can sometimes produce wrong values.

For platforms depending on the reflective implementation instead of the Unsafe implementation, this might be an unworkaroundable limitation. However, platforms that can very well use Unsafe should not be affected by that, too.

This means that the BinaryValueFunction lookup has to become a part of the modular MemoryAccessor logic so that the implementation can decide which strategy is best for it.

**Comment by tm-ms** [comment]
No. There is a universally correct solution.

For example for storing a reversed double value:
- there is a specific function STORE_double_REVERSED.
- it reads a double value from an instance.
- it converts the value to a raw long value. There is no conversion loss here.
- it then stores the long value itself to the store buffer, without conversion.

The inverse way:
- there is a specific function SETTER_double_REVERSED.
- it reads a long(!) value from the loading buffer.
- it reverts the long value. Now, the bits are in an order that is perfectly compatible with the desired double value.
- it converts the compatible raw long value to a double value
- it then sets the double value to the instance.

In other words: reversing logic never writes/reads floating point values to/from the target buffer.

**Comment by tm-ms** [comment]
Implemented the superior solution.
The initial solution has already been tested, so this issue is complete.
